webpack/assets/javascripts/react_app/redux/actions/common/forms.js

```diff
@@ -1,39 +1,51 @@
 import { APIActions } from '../../API';
 import { sprintf, translate as __ } from '../../../../react_app/common/I18n';
 
-export class SubmissionError {
-  constructor(errors) {
-    this.errors = errors;
-  }
-}
-
-const fieldErrors = ({ error }) => {
-  const { errors, severity } = error;
-
+const getBaseErrors = ({ error: { errors, severity } }) => {
+  let _error;
   if (errors.base) {
-    errors._error = {};
-    errors._error.errorMsgs = errors.base;
-    errors._error.severity = severity;
+    _error = {};
+    _error.errorMsgs = errors.base;
+    _error.severity = severity;
     delete errors.base;
   }
 
-  return new SubmissionError(errors);
+  return _error;
 };
 
-export const onError = error => {
+export const prepareErrors = (errors, base) =>
+  Object.keys(errors).reduce(
+    (memo, key) => {
+      const errorMessages = errors[key];
+
+      memo[key] =
+        errorMessages && errorMessages.join
+          ? errorMessages.join(', ')
+          : errorMessages;
+      return memo;
+    },
+    { _error: base }
+  );
+
+export const onError = (error, actions) => {
+  actions.setSubmitting(false);
   if (error.response.status === 422) {
-    // Handle invalid form data
-    throw fieldErrors(error.response.data);
+    const base = getBaseErrors(error?.response?.data);
+
+    actions.setErrors(
```

Comment:
there's no need to throw  errors, `actions` is a bag of formik's actions, including `setErrors`